### ISSUE TITLE
[3.0.4] image_tag recompiles all assets in test environment

### ISSUE THREAD

**Issue by ZipoKing** [rails/sprockets-rails#357]
Calling image_tag in test environment takes ages because all assets are trying to be recompiled. When I'll downgrade sprockets-rails gem to the latest 2.x version (2.3.3 now), everything works fine. I've killed Ruby process in the middle of the process and attaching below full backtrace since image_tag has been called (Ruby 2.3.1, Rails 4.2.6).
```
/usr/lib/ruby/gems/2.3.0/gems/actionview-4.2.6/lib/action_view/helpers/asset_tag_helper.rb:211:in `image_tag'
/usr/lib/ruby/gems/2.3.0/gems/actionview-4.2.6/lib/action_view/helpers/asset_url_helper.rb:292:in `image_path'
/usr/lib/ruby/gems/2.3.0/gems/actionview-4.2.6/lib/action_view/helpers/asset_url_helper.rb:135:in `asset_path'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-rails-3.0.4/lib/sprockets/rails/helper.rb:78:in `compute_asset_path'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-rails-3.0.4/lib/sprockets/rails/helper.rb:88:in `resolve_asset_path'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-rails-3.0.4/lib/sprockets/rails/helper.rb:228:in `resolve_asset'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-rails-3.0.4/lib/sprockets/rails/helper.rb:228:in `detect'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-rails-3.0.4/lib/sprockets/rails/helper.rb:228:in `each'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-rails-3.0.4/lib/sprockets/rails/helper.rb:229:in `block in resolve_asset'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-rails-3.0.4/lib/sprockets/rails/helper.rb:89:in `block in resolve_asset_path'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-rails-3.0.4/lib/sprockets/rails/helper.rb:311:in `asset_path'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-rails-3.0.4/lib/sprockets/rails/helper.rb:323:in `digest_path'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-rails-3.0.4/lib/sprockets/rails/helper.rb:349:in `raise_unless_precompiled_asset'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-rails-3.0.4/lib/sprockets/rails/helper.rb:345:in `precompiled?'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-rails-3.0.4/lib/sprockets/railtie.rb:214:in `block (3 levels) in <class:Railtie>'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-rails-3.0.4/lib/sprockets/railtie.rb:34:in `asset_precompiled?'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-rails-3.0.4/lib/sprockets/railtie.rb:49:in `precompiled_assets'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-rails-3.0.4/lib/sprockets/railtie.rb:49:in `map'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-rails-3.0.4/lib/sprockets/railtie.rb:49:in `each'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-3.6.0/lib/sprockets/manifest.rb:133:in `find'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-3.6.0/lib/sprockets/manifest.rb:133:in `each'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-3.6.0/lib/sprockets/manifest.rb:134:in `block in find'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-3.6.0/lib/sprockets/base.rb:73:in `find_all_linked_assets'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-3.6.0/lib/sprockets/base.rb:66:in `find_asset'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-3.6.0/lib/sprockets/cached_environment.rb:47:in `load'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-3.6.0/lib/sprockets/cached_environment.rb:20:in `block in initialize'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-3.6.0/lib/sprockets/loader.rb:44:in `load'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-3.6.0/lib/sprockets/loader.rb:317:in `fetch_asset_from_dependency_cache'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-3.6.0/lib/sprockets/loader.rb:60:in `block in load'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-3.6.0/lib/sprockets/loader.rb:134:in `load_from_unloaded'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-3.6.0/lib/sprockets/processor_utils.rb:56:in `call_processors'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-3.6.0/lib/sprockets/processor_utils.rb:56:in `reverse_each'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-3.6.0/lib/sprockets/processor_utils.rb:57:in `block in call_processors'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-3.6.0/lib/sprockets/processor_utils.rb:75:in `call_processor'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-3.6.0/lib/sprockets/sass_compressor.rb:28:in `call'
/usr/lib/ruby/gems/2.3.0/gems/sprockets-3.6.0/lib/sprockets/sass_compressor.rb:48:in `call'
/usr/lib/ruby/gems/2.3.0/gems/sass-3.4.22/lib/sass/engine.rb:281:in `render'
/usr/lib/ruby/gems/2.3.0/gems/sass-3.4.22/lib/sass/engine.rb:406:in `_to_tree'
/usr/lib/ruby/gems/2.3.0/gems/sass-3.4.22/lib/sass/scss/parser.rb:41:in `parse'
/usr/lib/ruby/gems/2.3.0/gems/sass-3.4.22/lib/sass/scss/parser.rb:125:in `stylesheet'
/usr/lib/ruby/gems/2.3.0/gems/sass-3.4.22/lib/sass/scss/parser.rb:667:in `block_contents'
/usr/lib/ruby/gems/2.3.0/gems/sass-3.4.22/lib/sass/scss/parser.rb:125:in `block in stylesheet'
/usr/lib/ruby/gems/2.3.0/gems/sass-3.4.22/lib/sass/scss/parser.rb:131:in `s'
/usr/lib/ruby/gems/2.3.0/gems/sass-3.4.22/lib/sass/scss/parser.rb:169:in `process_comment'
/usr/lib/ruby/gems/2.3.0/gems/sass-3.4.22/lib/sass/scss/parser.rb:169:in `[]'
```

**Comment by yivo** [comment]
Try to set `ils.application.config.assets.check_precompiled_asset` to `false`. By default all assets are compiled during start.

**Comment by rafaelfranca** [comment]
Thank you for the issue. To check is all assets you are using in your application are in the precompile list we need to precompile all the assets. Like @yivo said, if you don't want that behavior just disable it.

**Comment by ZipoKing** [comment]
@yivo @rafaelfranca Thanks for the answer, but two things in here:
* the newest available sprockets-rails gem version is 3.0.4 released in March (https://rubygems.org/gems/sprockets-rails)
* the option mentioned by you has been introduced in May, so it's not available in stable version

**Comment by rafaelfranca** [comment]
Yeah, you are correct, but is available in the master branch. While we don't release you can use it there.
